frontend/src/lib/WorkflowParser.ts

```diff
@@ -205,13 +206,19 @@ export default class WorkflowParser {
     if (selectedWorkflowNode && selectedWorkflowNode.outputs) {
       (selectedWorkflowNode.outputs.artifacts || [])
         .filter((a) => a.name === 'mlpipeline-ui-metadata' && !!a.s3)
-        .forEach((a) =>
+        .forEach((a) => {
+          let source = StorageService.MINIO;
+          if (a.s3!.endpoint.indexOf('s3.amazonaws.com') >= 0){
```

Comment:
Some tests are failing here and at the conditional below.
change to `if (a.s3!.endpoint && a.s3!.endpoint.indexOf(...) >= 0)`

Reply:
Thanks. I will make the change.

Reply:
There are more places: https://travis-ci.com/kubeflow/pipelines/jobs/197577227

Reply:
I think this would be removed. Let me run some tests on my end.